network/peer/tls_config_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package peer_test
```

Comment:
Out of curiosity, why is it beneficial to compile these as a testing package? Typically we've only done this to avoid import cycles.

Reply:
It's not necessary here. 

The `_test` can sometimes be useful because it forces the writer of the test to only use public APIs and not be able to look at internal state. 

I can change it to a non `_test` if it's an issue.